playbook/app/pb_kits/playbook/pb_passphrase/useZxcvbn.js

```diff
@@ -0,0 +1,54 @@
+import { useEffect, useMemo, useState } from 'react'
+import zxcvbn from 'zxcvbn'
+
+export default function useZxcvbn(options) {
+  const { passphrase = '', common, isPwned, confirmation, averageThreshold, minLength, strongThreshold } = options
+  const calculator = useMemo(
+    () => confirmation ?  () => ({ score: 0 }) : zxcvbn,
+    [confirmation]
+  )
+
+  const [percent, setPercent] = useState('0')
+  const [variant, setVariant] = useState('negative')
+  const [text, setText] = useState('\u00A0') //nbsp to keep height constant
+  const [result, setResult] = useState({})
+
+  useEffect(() => {
+    if (confirmation) {
+      return
+    }
+
+    setResult(calculator(passphrase))
+    const str = result.score
+
+    if (passphrase.length <= 0) {
+      setPercent('0')
+      setVariant('negative')
+      setText('\u00A0') //nbsp to keep height constant
+    } else if (common || isPwned) {
+      setPercent('25')
+      setVariant('negative')
+      setText('This passphrase is too common')
+    } else if (passphrase.length < minLength || str < averageThreshold) {
+      setPercent('25')
+      setVariant('negative')
+      setText('Too weak')
+    } else if (str < strongThreshold){
+      setPercent('50')
+      setVariant('warning')
+      setText('Almost there, keep going!')
+    } else if (str >= strongThreshold) {
+      setPercent('100')
+      setVariant('positive')
+      setText('Success! Strong passphrase')
+    }
```

Comment:
Not critical, but this area seems a little difficult to parse the logic. You could simplify it by taking the contents of the conditional statements and converting to human-readable variables. Take the two one for example:

```js
if (passphrase.length <= 0) {
...
} else if (common || isPwned) {
```

This could be:

```js
const noPassword = passphrase.length <= 0

if (noPassword) {
...
} else if (common || isPwned) {
...
```

and so on. For me, this makes reading the logic quick and decisive.

Reply:
```suggestion
    if (noPassword) {
      setPercent('0')
      setVariant('negative')
      setText('\u00A0') //nbsp to keep height constant
    } else if (commonPassword) {
      setPercent('25')
      setVariant('negative')
      setText('This passphrase is too common')
    } else if (weakPassword) {
      setPercent('25')
      setVariant('negative')
      setText('Too weak')
    } else if (averagePassword){
      setPercent('50')
      setVariant('warning')
      setText('Almost there, keep going!')
    } else if (strongPassword) {
      setPercent('100')
      setVariant('positive')
      setText('Success! Strong passphrase')
    }
```

Just an example from above

Reply:
Updated as suggested. Definitely easier to parse at a glance